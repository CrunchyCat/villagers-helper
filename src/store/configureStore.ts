import { configureStore, StoreEnhancer } from '@reduxjs/toolkit'
import { createInjectorsEnhancer } from 'redux-injectors'
import createSagaMiddleware from 'redux-saga'
import { createReducer } from './reducers'

export const configureAppStore = () => {
  const sagaMiddleware = createSagaMiddleware()
  const { run: runSaga } = sagaMiddleware

  const enhancers = [
    createInjectorsEnhancer({
      createReducer,
      runSaga
    })
  ] as StoreEnhancer[]

  // Return store with middlewares and enhancers
  return configureStore({
    reducer: createReducer(),
    middleware: defaultMiddleware => [...defaultMiddleware(), sagaMiddleware],
    devTools: process.env.NODE_ENV !== 'production',
    enhancers
  })
}
