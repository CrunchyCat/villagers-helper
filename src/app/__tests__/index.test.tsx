import { Provider } from 'react-redux'
import { createRenderer } from 'react-test-renderer/shallow'
import { configureAppStore } from 'store/configureStore'
import { App } from '../index'

const renderer = createRenderer()

const AppWrapper = () => {
  const store = configureAppStore()

  return (
    <Provider store={store}>
      <App />
    </Provider>
  )
}

describe('<App />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<AppWrapper />)
    const renderedOutput = renderer.getRenderOutput()
    expect(renderedOutput).toMatchSnapshot()
  })
})
