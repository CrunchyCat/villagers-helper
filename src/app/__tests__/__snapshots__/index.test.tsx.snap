// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "createReducer": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
    }
  }
>
  <App />
</Provider>
`;
